src/client/pmix_client.c

```diff
@@ -236,6 +236,24 @@ static void evhandler_reg_callbk(pmix_status_t status,
     *active = status;
 }
 
+static void _destruct_my_server_fn(int sd, short args, void *cbdata)
+{
+    bool *active = (bool *)cbdata;
+    PMIX_DESTRUCT(&pmix_client_globals.myserver);
+    *active = false;
+}
+
+
+static void pmix_destruct_my_server(void)
+{
+    pmix_event_t ev;
+    volatile bool active = true;
+    event_assign(&ev, pmix_globals.evbase, -1,
```

Comment:
We have a macro for that:
https://github.com/pmix/master/blob/master/src/client/pmix_client_get.c#L179